libraries/AP_VisualOdom/AP_VisualOdom.h

```diff
@@ -92,7 +93,10 @@ class AP_VisualOdom
     // distances in meters, roll, pitch and yaw are in radians
     void handle_vision_position_estimate(uint64_t remote_time_us, uint32_t time_ms, float x, float y, float z, float roll, float pitch, float yaw, float posErr, float angErr, uint8_t reset_counter);
     void handle_vision_position_estimate(uint64_t remote_time_us, uint32_t time_ms, float x, float y, float z, const Quaternion &attitude, float posErr, float angErr, uint8_t reset_counter);
-    
+    // This override currently narrows to 32-bit precision regardless of EKF using double math.
+    // AP_VisualOdom does not yet support double math.
+    // When it does support double math, the function signature can remain the same.
```

Comment:
Well, the `posErr` and `angErr` fields should be converted to doubles, but cartographer doesn't supply those. I'd be interested to look at some of the other backends to see if they would make use of double precision. For GSoC, we decided using the existing single-precision localization is fine.

Reply:
We hard code these as zero because cartographer doesn't supply them.